yarn-project/end-to-end/src/e2e_multi_transfer.test.ts

```diff
@@ -0,0 +1,119 @@
+    ownerAddress = accounts[0];
+
+    for (let i = 1; i < accounts.length; i++) {
+      const account = accounts[i];
+      recipients.push(account);
+    }
+  }, 100_000);
+
+  afterEach(async () => {
+    await aztecNode?.stop();
+    if (aztecRpcServer instanceof AztecRPCServer) {
+      await aztecRpcServer?.stop();
+    }
+  }, 30_000);
+
+  const deployZkTokenContract = async (initialBalance: bigint, owner: AztecAddress) => {
+    logger(`Deploying zk token contract...`);
+    zkTokenContract = await PrivateTokenAirdropContract.deploy(wallet, initialBalance, owner).send().deployed();
+    logger(`zk token contract deployed at ${zkTokenContract.address}`);
+  };
+
+  const deployMultiTransferContract = async () => {
+    logger(`Deploying multi-transfer contract...`);
+    multiTransferContract = await MultiTransferContract.deploy(wallet).send().deployed();
+    logger(`multi-transfer contract deployed at ${multiTransferContract.address}`);
+  };
+
+  const expectBalance = async (tokenContract: Contract, owner: AztecAddress, expectedBalance: bigint) => {
+    const [balance] = await tokenContract.methods.getBalance(owner).view({ from: owner });
+    logger(`Account ${owner} balance: ${balance}`);
+    expect(balance).toBe(expectedBalance);
+  };
+
+  it('12 transfers per transactions should work', async () => {
+    const initialNote = 1000n;
+
+    logger(`Deploying zk token contract...`);
+    await deployZkTokenContract(initialNote, ownerAddress);
+
+    logger(`Deploying multi-transfer contract...`);
+    await deployMultiTransferContract();
+
+    // owner: 1000 => [100, 200, 300, 400]
+    logger(`self batchTransfer()`);
+    const batchTransferTx = zkTokenContract.methods
+      .batchTransfer(ownerAddress, [200n, 300n, 400n], [ownerAddress, ownerAddress, ownerAddress], 0, 0)
+      .send({ origin: ownerAddress });
+    await batchTransferTx.isMined();
+    const batchTransferTxReceipt = await batchTransferTx.getReceipt();
+    logger(`consumption Receipt status: ${batchTransferTxReceipt.status}`);
+    await expectBalance(zkTokenContract, ownerAddress, initialNote);
+    await expectsNumOfEncryptedLogsInTheLastBlockToBe(aztecNode, 4);
+
+    const amounts: bigint[] = [20n, 25n, 30n, 40n, 50n, 60n, 75n, 80n, 85n, 100n, 120n, 130n];
+    const amountSum = amounts.reduce((a, b) => a + b, 0n);
+
+    /**
+     * owner: [100, 200, 300, 400]
+     *         |    |    |    |
+     *         |    |    |    [50 (o), 100, 120, 130] batchTx
+     *         |    |    |
+     *         |    |    [60 (o), 75, 80, 85] batchTx
+     *         |    |
+     *         |    [50 (o), 40, 50, 60] batchTx
+     *         |
+     *         [25 (o), 20, 25, 30] batchTx
+     *
+     * o = owner
+     */
+    logger(`multiTransfer()...`);
+    const multiTransferTx = multiTransferContract.methods
+      .multiTransfer(
+        zkTokenContract.address.toField(),
+        recipients.concat(recipients),
```

Comment:
add a comment saying insteda of creating 12 accounts, we just send to each account twice

Reply:
created 12 accounts now that it doesn't take n blocks to create n accounts.